IdentityCore/src/cache/token/MSIDCommonCredential.h

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#import <Foundation/Foundation.h>
+#import "MSIDCredentialType.h"
+#import "MSIDJsonSerializable.h"
+
+@class MSIDClientInfo;
+
+@interface MSIDCommonCredential : NSObject <NSCopying, MSIDJsonSerializable>
```

Comment:
When MSIDCommonCredential as a temporary solution would I work, in long-term I suggest to not add "Common" word to class name if it is not necessary. In this particular case, I think we should change current MSIDCredentialCacheItem instead of introducing a new class.

Reply:
It's my understanding that that type of refactoring is for later. We need to unblock our project with a handful of hopefully shareable classes for now, minimizing our impact on the msal-objc devs & codebase.

Reply:
Sounds good, just wanted to double check that in long-term perspective we will refactor the code and remove "Common" classes.

Reply:
If that's not the end result, this code shouldn't be merged to dev, until ready.